modules/apps/commerce/commerce-payment-service/src/main/java/com/liferay/commerce/payment/internal/engine/CommercePaymentEngineImpl.java

```diff
@@ -540,8 +544,16 @@ private List<CommercePaymentMethod> _getCommercePaymentMethodsList(
 				_commercePaymentMethodRegistry.getCommercePaymentMethod(
 					commercePaymentMethodGroupRel.getEngineKey());
 
-			if (subscriptionOrder &&
-				!commercePaymentMethod.isProcessRecurringEnabled()) {
+			boolean hasPermission = permissionChecker.hasPermission(
+				commercePaymentMethodGroupRel.getGroupId(),
+				CommercePaymentMethodGroupRel.class.getName(),
```

Comment:
@dsitu can you have a look at this? I don't remember if Bchan accept this. 
Thanks!